File: frontend/src/api/experience.ts
/**
 * 业务履历管理API
 */
import request from './index'

export interface Experience {
  id: number
  user_id: number
  project: string
  module?: string
  role?: string
  description?: string
  man_days: number
  created_at: string
  updated_at: string
}

export interface ExperienceCreate {
  project: string
  module?: string
  role?: string
  description?: string
  man_days: number
}

export interface ExperienceUpdate {
  project?: string
  module?: string
  role?: string
  description?: string
  man_days?: number
}

export interface ExperienceListResponse {
  total: number
  items: Experience[]
}

/**
 * 创建业务履历
 */
export function createExperience(data: ExperienceCreate): Promise<Experience> {
  return request.post('/api/v1/experiences', data)
}

/**
 * 获取业务履历列表
 */
export function getExperiences(): Promise<ExperienceListResponse> {
  return request.get('/api/v1/experiences')
}

/**
 * 获取业务履历详情
 */
export function getExperience(experienceId: number): Promise<Experience> {
  return request.get(`/api/v1/experiences/${experienceId}`)
}

/**
 * 更新业务履历
 */
export function updateExperience(experienceId: number, data: ExperienceUpdate): Promise<Experience> {
  return request.put(`/api/v1/experiences/${experienceId}`, data)
}

/**
 * 删除业务履历
 */
export function deleteExperience(experienceId: number): Promise<void> {
  return request.delete(`/api/v1/experiences/${experienceId}`)
}

File: backend/app/schemas/experience.py
"""业务履历相关模式"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from decimal import Decimal


class ExperienceBase(BaseModel):
    """业务履历基础模式"""
    project: str = Field(..., min_length=1, max_length=100, description="项目名称")
    module: Optional[str] = Field(None, max_length=100, description="模块名称")
    role: Optional[str] = Field(None, max_length=50, description="角色")
    description: Optional[str] = Field(None, description="贡献描述")
    man_days: Decimal = Field(..., ge=0, description="投入人天，支持小数（如0.5天）")


class ExperienceCreate(ExperienceBase):
    """创建业务履历请求"""
    pass


class ExperienceUpdate(BaseModel):
    """更新业务履历请求"""
    project: Optional[str] = Field(None, min_length=1, max_length=100, description="项目名称")
    module: Optional[str] = Field(None, max_length=100, description="模块名称")
    role: Optional[str] = Field(None, max_length=50, description="角色")
    description: Optional[str] = Field(None, description="贡献描述")
    man_days: Optional[Decimal] = Field(None, ge=0, description="投入人天")


class ExperienceResponse(BaseModel):
    """业务履历响应"""
    id: int
    user_id: int
    project: str
    module: Optional[str]
    role: Optional[str]
    description: Optional[str]
    man_days: Decimal
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ExperienceListResponse(BaseModel):
    """业务履历列表响应"""
    total: int
    items: list[ExperienceResponse]
